torch/distributed/_spmd/api.py

```diff
@@ -578,7 +537,12 @@ def wrapper(*args, **kwargs):
             compiled_obj = wrapper.__dict__.get(COMPILED_OBJECT_KEY, None)
             if compiled_obj is None:
                 first_iter = True
-                compiled_obj = _compile(func, module_override, *args, **kwargs)
+                global dtensor_expand_mode
```

Comment:
(no need to change this PR) curious why do create `dtensor_expand_mode` as a global instead of creating a new one for every recompilation. Do we expect we need pass states across recompilations or will the constructor becoming more expensive later on?

Reply:
The main reason is that there's a `_override_placements` API that uses `_placements_override` (i think it's for sparse part integration), where `DTensorExpandMode` put this as an attribute, so it need to be a global var now in order for the `_override_placements` to still work, we might can put this into `DTensorExpand` mode too but that would also be BC breaking.